plugins/aws/mq/mqBrokerEncrypted.spec.js

```diff
@@ -0,0 +1,229 @@
+            "AWSAccountId": "000011112222",
+            "KeyId": "c4750c1a-72e5-4d16-bc72-0e7b559e0250",
+            "Arn": "arn:aws:kms:us-east-1:000011112222:key/c4750c1a-72e5-4d16-bc72-0e7b559e0250",
+            "CreationDate": "2020-12-15T01:16:53.045000+05:00",
+            "Enabled": true,
+            "Description": "Default master key that protects my Glue data when no other key is defined",
+            "KeyUsage": "ENCRYPT_DECRYPT",
+            "KeyState": "Enabled",
+            "Origin": "AWS_KMS",
+            "KeyManager": "AWS",
+            "CustomerMasterKeySpec": "SYMMETRIC_DEFAULT",
+            "EncryptionAlgorithms": [
+                "SYMMETRIC_DEFAULT"
+            ]
+        }
+    }
+];
+
+const listKeys = [
+    {
+        "KeyId": "0604091b-8c1b-4a55-a844-8cc8ab1834d9",
+        "KeyArn": "arn:aws:kms:us-east-1:000011112222:key/c4750c1a-72e5-4d16-bc72-0e7b559e0250"
+    }
+]
+
+const createCache = (brokers, keys, describeBroker, describeKey, brokersErr, keysErr, describeKeyErr, describeBrokerErr) => {
+    var keyId = (keys && keys.length && keys[0].KmsKeyId) ? keys[0].KmsKeyId : null;
+    var BrokerId = (brokers && brokers.length) ? brokers[0].BrokerId: null;
+    return {
+        mq: {
+            listBrokers: {
+                'us-east-1': {
+                    err: brokersErr,
+                    data: brokers
+                },
+            },
+            describeBroker: {
+                'us-east-1': {
+                    [BrokerId]: {
+                        data: describeBroker,
+                        err: describeBrokerErr
+                    }
+                }
+            }
+        },
+        kms: {
+            listKeys: {
+                'us-east-1': {
+                    data: keys,
+                    err: keysErr
+                }
+            },
+            describeKey: {
+                'us-east-1': {
+                    [keyId]: {
+                        err: describeKeyErr,
+                        data: describeKey
+                    },
+                },
+            },
+        },
+    };
+};
+
+describe('mqBrokerEncrypted', function () {
+    describe('run', function () {
+        it('should PASS if MQ Broker data at-rest is encrypted with desired encryption level', function (done) {
+            const cache = createCache(listBrokers[0], listKeys, [describeBroker[1]], describeKey[0]);
+            mqBrokerEncrypted.run(cache, { mq_broker_desired_encryption_level : 'awscmk' }, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should FAIL if MQ Broker data at-rest is not encrypted with desired encryption level', function (done) {
+            const cache = createCache([listBrokers[1]],listKeys, [describeBroker[0]], describeKey[1]);
```

Comment:
describeBroker[0] has a different key which is not there in describeKey or listKeys. is this test working as expected if so how?